src/todo.ts

```diff
@@ -0,0 +1,99 @@
+import { Item } from './item';
+class TodoApp {
+    /**
+     * Allows the creation and maintenance of a simple todo list
+     */
+
+    public toDoList: Map<number, Item>;
+    public curId: number;
+
+    constructor() {
+        this.curId = 1;
+        this.toDoList = new Map<number, Item>();
+    }
+
+    // Had to generate the item ID here so two todoApps could have separate id lists
+    public addItem(description: string, dueAt: string): Item {
+        const date = new Date(dueAt);
```

Comment:
So why not make `dueAt` a Date instead of parsing?  Maybe I missed a requirement?

Reply:
The due date of each item enters the system as a string, so it has to get converted to a Date object somewhere. I chose to do it here rather than in server.ts so the todo.ts could still be used as a terminal application if needed.